### DISCUSSION TITLE
How to use `graphql-ws` protocol?

### DISCUSSION THREAD

**Discussion by bitttttten** [enisdenjo/graphql-ws#519] [Q&A] [answered]
I see in the code that the protocol is hard coded to "graphql-transport-ws". We use a GraphQL server that is setup with "graphql-ws". Is there any way to configure the library to allow the deprecated protocol?

Referring also to this line: https://github.com/enisdenjo/graphql-ws/blob/b3477570c85537c1ea6bb394bdd89b2a9e7c8482/src/client.ts#L643

```
          const socket = new WebSocketImpl(
            typeof url === 'function' ? await url() : url,
            GRAPHQL_TRANSPORT_WS_PROTOCOL,
          );

```

It doesn't seem like it's able to be configured from our side 😅 

Thanks!

**Comment by enisdenjo** [comment, marked as answer]
Sorry to disappoint; but this library is not, and will never be, backwards compatible with the [deprecated "graphql-ws" protocol from `subscriptions-transport-ws`](https://github.com/apollographql/subscriptions-transport-ws). Also stated in [readme's disclamer](https://github.com/enisdenjo/graphql-ws#disclaimer).

You must upgrade the server to use this library too. If your server is in JavaScript, that shouldn't be a big deal! Happy to help.
